src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1001,7 +975,7 @@ public class ForkJoinPool extends AbstractExecutorService {
      * Initial capacity of work-stealing queue array.  Must be a power
      * of two, at least 2. See above.
      */
-    static final int INITIAL_QUEUE_CAPACITY = 1 << 6;
+    static final int INITIAL_QUEUE_CAPACITY = 1 << 7;
```

Comment:
@DougLea Interesting change—what impact did you notice when doubling this?

Reply:
After recheckiing, the best policy is to leave internal queues the same, but initialize external queues larger.

Reply:
Probably a misplaced post again, but why can't you allocate an array of arrays
where the outer array is 30
and the inner array will be doubled, 
then the segment can be calculated by log(index)
with: resize O(1), read/write O(1).

This approach can also be used in ConcurrentHashMap
sample implementation 
https://github.com/sunmisc/wormhole/blob/main/main/src/main/java/sunmisc/utils/concurrent/memory/ReferenceSegmentMemory.java

I don't know if this method exists anywhere else, I invented it, maybe it was invented before me.

Reply:
Thanks for link to sample code link. I had tried something like this in an analogous case in SubmissionPublisher, but couldn't reduce overhead enough to be worthwhile. I may try again there first,m since resizing there is more common and disruptive.